src/kktsystem.jl

```diff
@@ -313,3 +392,145 @@ function initialize!(kkt::SparseUnreducedKKTSystem)
 end
 
 is_reduced(::SparseUnreducedKKTSystem) = false
+
+#=
+    DenseKKTSystem
+=#
+
+struct DenseKKTSystem{T, VT, MT} <: AbstractKKTSystem{T, MT}
+    hess::MT
+    jac::MT
```

Comment:
Didn't realize this in my first review, but can't these be views so that we don't need to transfer this in `_build_dense_kkt_system`?